### PR TITLE
[ENH] Make explicit that "task" metadata applies to "beh" modality

### LINKED ISSUE #795: "standardization" of task metadata across modality
Was having a look at the ephys BEP and realized that even though this is a modality that uses the task entity the BEP does not menton some of the typical task related fields that other modalities have.

But I then realized we do have this problem in the current spec already ( see below).

So I would suggest to make more consistent what task-related fields are available for all modalities.

The fields in question are the following (I may have missed some):

```
      "Instructions": "RECOMMENDED",
      "TaskDescription": "RECOMMENDED",
      "CogAtlasID": "RECOMMENDED",
      "CogPOID": "RECOMMENDED",
```

And they not mentioned for :
- task based PET
- behavioral

I suggest opening a PR to add those.

---

This also highlight the need to do some refactoring of some of the task aspects that are common across modalities.

Something that is related to what @robertoostenveld mentioned in #575 

After adding those I suggest, we should probably find a way to do this refactoring.

### PR DESCRIPTION
**Pull request by Remi-Gau**
Fixes #795 

Adds task metadata that exist in other modalities to:
- behavioral
- ~~PET~~

All metadata is RECOMMENDED. 
Making `TaskName` REQUIRED (like it is for other modalities) could break backward compatibility, no?

I used a pretty global approach for `beh` assuming that these metadata could go in any of JSON flles on can find in this folder: should we have a more fine grain approach to this?

---

- [ ] update validator according to this PR (see [this comment](https://github.com/bids-standard/bids-specification/pull/804#issuecomment-851249374))

### CODE REVIEW

**Comment by Remi-Gau** [comment]
@melanieganz and @mnoergaard I am adding you as reviewers as this also relates to PET. :smile:

File: src/04-modality-specific-files/07-behavioral-experiments.md
Comment (mnoergaard) [inline src/04-modality-specific-files/07-behavioral-experiments.md:28]: Maybe this should be made more clear that it addresses task data

```suggestion
## RECOMMENDED task metadata
```

File: src/04-modality-specific-files/09-positron-emission-tomography.md
Comment (mnoergaard) [inline src/04-modality-specific-files/09-positron-emission-tomography.md:226]: I would say that 99.9% of PET studies don't carry out a task (e.g. combined PET and fMRI). Therefore, this seems to be taking a lot of space, and may potentially confuse PET people. Wouldn't it be possible in the task section in the top to refer to this metadata in the behavioral experiments?

Comment (Remi-Gau) [reply src/04-modality-specific-files/09-positron-emission-tomography.md:226]: Actually forgot to mention but at our last at our last maintainers meeting we said that we'll this out of PET for now: the long term plan would to refactor task related stuff into a separate sections that we can "point" to from all concerned modalities. 

I would then get added to PET at this time.

So I will remove it from PET in this PR for now.

**Review by sappelhoff — COMMENTED** [review]
LGTM, probably something to remind ourselves of in https://github.com/bids-standard/bids-specification/pull/774/

**Comment by Remi-Gau** [comment]
> LGTM, probably something to remind ourselves of in #774

Good point.

Once this is merged I will update #774 (and also the validator ?)

**Comment by sappelhoff** [comment]
Mmh yes ... about updating the validator: Would you add it to the `events.json` schema, or the `beh.json` schema? The latter doesn't exist yet, but I think it'd be a better choice, and also something to clarify in this PR. WDYT?

Otherwise we'd have to specify it in the events.json schema: https://github.com/bids-standard/bids-validator/blob/master/bids-validator/validators/json/schemas/events.json

**Comment by Remi-Gau** [comment]
I would be more in favor of creating a `beh.json` schema.

This will need to happen at some point anyway.

File: src/04-modality-specific-files/07-behavioral-experiments.md
Comment (effigies) [inline src/04-modality-specific-files/07-behavioral-experiments.md:36]: Rest doesn't really apply for `beh`.
```suggestion
| TaskName           | RECOMMENDED           | [string][]    | Name of the task. No two tasks should have the same name. The task label included in the file name is derived from this `TaskName` field by removing all non-alphanumeric (`[a-zA-Z0-9]`) characters. For example `TaskName` `"faces n-back"` will correspond to task label `facesnback`. |
```

File: src/04-modality-specific-files/07-behavioral-experiments.md
Comment (effigies) [inline src/04-modality-specific-files/07-behavioral-experiments.md:39]:
```suggestion
| CogAtlasID         | RECOMMENDED           | [string][]    | [URI][uri] of the corresponding [Cognitive Atlas](https://www.cognitiveatlas.org/) term that describes the task (for example, Stroop task "<https://www.cognitiveatlas.org/task/id/tsk_4a57abb949e27>").                                                                                                                                                          |
```

File: src/04-modality-specific-files/07-behavioral-experiments.md
Comment (effigies) [inline src/04-modality-specific-files/07-behavioral-experiments.md:40]:
```suggestion
| CogPOID            | RECOMMENDED           | [string][]    | [URI][uri] of the corresponding [CogPO](http://www.cogpo.org/) term that describes the task (for example, Rest "<http://wiki.cogpo.org/index.php?title=Stroop_Task_Paradigm>") .                                                                                                                                                                                                                     |
```

**Review by effigies — APPROVED** [review]
LGTM. Suggested switching to Stroop instead of rest, since rest produces no behavioral data.

**Review by sappelhoff — CHANGES_REQUESTED** [review]
This PR will have to be updated with respect to #774 --> metadata tables are now rendered via macros directly from the schema.

Other than that technical change request, I still approve of this PR

File: src/04-modality-specific-files/07-behavioral-experiments.md
Comment (Remi-Gau) [inline src/04-modality-specific-files/07-behavioral-experiments.md:40]:
```suggestion
| CogPOID            | RECOMMENDED           | [string][]    | [URI][uri] of the corresponding [CogPO](http://www.cogpo.org/) term that describes the task (for example, Stroop "<http://wiki.cogpo.org/index.php?title=Stroop_Task_Paradigm>") .                                                                                                                                                                                                                     |
```

File: src/04-modality-specific-files/07-behavioral-experiments.md
Comment (Remi-Gau) [reply src/04-modality-specific-files/07-behavioral-experiments.md]: just replacing your suggestion with one that changes rest to stroop in text **and** URL.

File: src/04-modality-specific-files/07-behavioral-experiments.md
Comment (effigies) [inline src/04-modality-specific-files/07-behavioral-experiments.md:42]: Mending fences.

```suggestion
| **Key name**       | **Requirement level** | **Data type** | **Description**
                                                               |
| ------------------ | --------------------- | ------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| TaskName           | RECOMMENDED           | [string][]    | Name of the task. No two tasks should have the same name. The task label included in the file name is derived from this `TaskName` field by removing all non-alphanumeric (`[a-zA-Z0-9]`) characters. For example `TaskName` `"faces n-back"` will correspond to task label `facesnback`. |
| TaskDescription    | RECOMMENDED           | [string][]    | Description of the task.                                                                                                                                                                                                                                                                  |
| Instructions       | RECOMMENDED           | [string][]    | Text of the instructions given to participants before the scan. This is not only important for behavioral or cognitive tasks but also in resting state paradigms (for example, to distinguish between eyes open and eyes closed).                                                         |
| CogAtlasID         | RECOMMENDED           | [string][]    | [URI][uri] of the corresponding [Cognitive Atlas](https://www.cognitiveatlas.org/) term that describes the task (for example, Stroop task "<https://www.cognitiveatlas.org/task/id/tsk_4a57abb949e27>").                                                                                  |
| CogPOID            | RECOMMENDED           | [string][]    | [URI][uri] of the corresponding [CogPO](http://www.cogpo.org/) term that describes the task (for example, Stroop "<http://wiki.cogpo.org/index.php?title=Stroop_Task_Paradigm>") .                                                                                                        |
| InstitutionName    | RECOMMENDED           | [string][]    | The name of the institution in charge of the equipment that produced the composite instances.                                                                                                                                                                                             |
| InstitutionAddress | RECOMMENDED           | [string][]    | The address of the institution in charge of the equipment that produced the composite instances.                                                                                                                                                                                          |
```

**Comment by Remi-Gau** [comment]
> This PR will have to be updated with respect to #774 --> metadata tables are now rendered via macros directly from the schema.

Updated the PR to use the macros and the schema definitions. 

The problem is that now the definitions for `TaskName` and `Instructions` both make reference to resting state which does not apply to behavior (see @effigies comments above).

Suggestions: shorten the base definitions and add the "resting state" mentions where necessary (func, eeg, meg, ieeg).

@tsalo does that make sense to you?

**Comment by tsalo** [comment]
:+1: That works for me!

File: src/04-modality-specific-files/04-intracranial-electroencephalography.md
Comment (Remi-Gau) [inline src/04-modality-specific-files/04-intracranial-electroencephalography.md:101]: I took the liberty to remove this sentence from the iEEG page as its content is not specific to iEEG. Happy to leave it in or better add to the definitions for fMRI, iEEG, MEG, EEG.

**Comment by Remi-Gau** [comment]
Tempted to add some info in the contributing.md on how to formulate the call to the macros to use the metadata description contained in the schema. 
Just so that people are aware that those base description can be extended.

File: src/04-modality-specific-files/04-intracranial-electroencephalography.md
Comment (sappelhoff) [reply src/04-modality-specific-files/04-intracranial-electroencephalography.md:101]: I think it's in the iEEG spec because of the large amount of iEEG datasets with patients not doing anything in particular (not even deliberate rest) ... for example because everybody is waiting for the next onset of a seizure. It may be that a better place for this info is in the starter kit though ...

I am not opposed to removing it here as long as we have it **somwhere**

**Comment by sappelhoff** [comment]
> Tempted to add some info in the contributing.md on how to formulate the call to the macros to use the metadata description contained in the schema.
Just so that people are aware that those base description can be extended.

I think that's a good idea 👍  we'll have to refer people to such docs quite some times in the future

**Review by sappelhoff — APPROVED** [review]

File: src/04-modality-specific-files/04-intracranial-electroencephalography.md
Comment (Remi-Gau) [reply src/04-modality-specific-files/04-intracranial-electroencephalography.md:101]: hum... that makes sense that for this use case this should be emphasized here. Let's keep it in for now.

Comment (effigies) [reply src/04-modality-specific-files/04-intracranial-electroencephalography.md:101]: Agree with reverting this deletion. Otherwise +1 to merge.

**Comment by sappelhoff** [comment]
Thanks @Remi-Gau :tada: let's get the validator PR fixed and merged next

**Comment by Remi-Gau** [comment]
yup.
in the meantime I will update #839

**Pull request merged**
